### PR TITLE
Fix FreeBSD Mem%

### LINKED ISSUE #373: FreeBSD MEM% 0
All percentages in PERCENT_MEM column are always 0%.

FreeBSD 10.2-RELEASE-p9, htop 2.0.0

### PR DESCRIPTION
**Pull request by Sp1l**
Imported from the FreeBSD port https://svnweb.freebsd.org/ports/head/sysutils/htop/files/patch-freebsd__FreeBSDProcessList.c?view=markup&pathrev=408842

Fixes #373

### CODE REVIEW

**Comment by Sp1l** [comment]
Verified
```
  PID USER      PRI  NI  VIRT   RES S CPU% MEM%   TIME+  Command
 2254 root       80   0  130M  115M R 28.2  1.4  0:03.33 /usr/bin/c++ -cc1 -tri...
 2648 root       76   0 75396 61348 R 13.4  0.8  0:01.41 /usr/bin/c++ -cc1 -tri...
```

**Comment by Sp1l** [comment]
Fixes #400 as well

File: freebsd/FreeBSDProcessList.c [hunk 1]
```diff
@@ -479,6 +483,7 @@ void ProcessList_goThroughEntries(ProcessList* this) {
       // from FreeBSD source /src/usr.bin/top/machine.c
       proc->m_size = kproc->ki_size / 1024 / pageSizeKb;
       proc->m_resident = kproc->ki_rssize;
+      proc->percent_mem = (proc->m_resident * PAGE_SIZE_KB) / (double)(Global_totalMem) * 100.0;
```
Comment (hishamhm) [inline freebsd/FreeBSDProcessList.c]: why the global, and not just `this->totalMem`?

**Comment by hishamhm** [comment]
Thank you for the fixes! Please check my comment above and test it with the suggested change.

**Comment by hishamhm** [comment]
Just noticed the change I suggested is what #391 does, btw!

File: freebsd/FreeBSDProcessList.c (hunk 1 above)
Comment (gaod) [inline freebsd/FreeBSDProcessList.c]: I have the same question :~

**Pull request merged**